homeassistant/components/epion/sensor.py

```diff
@@ -0,0 +1,127 @@
+"""Support for Epion API."""
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_MILLION,
+    PERCENTAGE,
+    UnitOfPressure,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import generate_entity_id
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_API_CLIENT, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add an Epion entry."""
+    epion_base = hass.data[DOMAIN][entry.entry_id][DATA_API_CLIENT]
+
+    entities = []
+    current_data = epion_base.last_response
+    for epion_device in current_data["devices"]:
+        # Relevant keys are: deviceId, deviceName, locationId, lastMeasurement, co2, temperature, humidity, pressure
+        entities.append(EpionSensor(epion_base, epion_device, "co2", hass))
+        entities.append(EpionSensor(epion_base, epion_device, "temperature", hass))
+        entities.append(EpionSensor(epion_base, epion_device, "humidity", hass))
+        entities.append(EpionSensor(epion_base, epion_device, "pressure", hass))
+
+    async_add_entities(entities)
+
+
+class EpionSensor(SensorEntity):
+    """Representation of an Epion Air sensor."""
+
+    def __init__(self, epion_base, epion_device, key, hass) -> None:
+        """Initialize an EpionSensor."""
+        self._epion_base = epion_base
+        self._epion_device = epion_device
+        self._measurement_key = key
+        self._last_value: float | None = self.extract_value()
+        self._display_name = ""
+        if key == "temperature":
+            self._attr_native_unit_of_measurement = UnitOfTemperature.CELSIUS
+            self._attr_device_class = SensorDeviceClass.TEMPERATURE
+            self._display_name = "Temperature"
+        elif key == "humidity":
+            self._attr_native_unit_of_measurement = PERCENTAGE
+            self._attr_device_class = SensorDeviceClass.HUMIDITY
+            self._display_name = "Humidity"
+        elif key == "pressure":
+            self._attr_native_unit_of_measurement = UnitOfPressure.HPA
+            self._attr_device_class = SensorDeviceClass.ATMOSPHERIC_PRESSURE
+            self._display_name = "Pressure"
+        else:
+            self._attr_native_unit_of_measurement = CONCENTRATION_PARTS_PER_MILLION
+            self._attr_device_class = SensorDeviceClass.CO2
+            self._display_name = "CO2"
```

Comment:
Please look into `SensorEntityDescription`s, they will make this a lot easier

Reply:
Updated to use a sequence of `SensorEntityDescription`, also bringing in the suggested precision through there per your last comment.